app/scripts/modules/core/pipeline/config/triggers/docker/dockerTrigger.module.js

```diff
@@ -48,15 +48,20 @@ module.exports = angular.module('spinnaker.core.pipeline.trigger.docker', [
       }
       let key = `${trigger.registry}/${trigger.repository}`;
       $scope.tags = $scope.repositoryMap[key] || [];
-      if ($scope.tags.indexOf(trigger.tag) < 0) {
-        trigger.tag = null;
-      }
     }
 
     this.clearTag = function() {
       trigger.tag = null;
     };
 
+    $scope.getTags = function(search) {
+      var newOne = $scope.tags.slice();
+      if (search && $scope.tags.slice().indexOf(search) === -1) {
+        newOne.unshift(search);
```

Comment:
I'm by no means a javascript expert, but won't this add a new item to the list of tags everytime the query is updated (for each keystroke)?